controllers/replication.storage/volumereplication_controller.go

```diff
@@ -363,16 +365,51 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	}
 
 	instance.Status.LastCompletionTime = getCurrentTime()
+
+	var requeueForInfo bool
+
+	if instance.Spec.ReplicationState == replicationv1alpha1.Primary {
+		info, err := r.getLastSyncTime(vr)
+		if err != nil {
+			logger.Error(err, "Failed to get volume replication info")
+			return ctrl.Result{}, err
+		}
+		lastSynctime := metav1.NewTime(info.AsTime())
+		instance.Status.LastSyncTime = &lastSynctime
+		requeueForInfo = true
+	}
 	err = r.updateReplicationStatus(instance, logger, getReplicationState(instance), msg)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
 	logger.Info(msg)
 
+	scheduleTime := getScheduleTime(parameters, logger)
+	if requeueForInfo {
+		logger.Info("volume is primary, requeuing to get volume replication info")
```

Comment:
logging is not much helpful. Remove it on add the next queue interval time. and also move line 388 inside the if condition